opentracing-examples/src/test/java/io/opentracing/examples/late_span_finish/TestLateSpanFinish.java

```diff
@@ -0,0 +1,93 @@
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package io.opentracing.examples.late_span_finish;
+
+import io.opentracing.ActiveSpan;
+import io.opentracing.Span;
+import io.opentracing.mock.MockSpan;
+import io.opentracing.mock.MockTracer;
+import io.opentracing.mock.MockTracer.Propagator;
+import io.opentracing.util.ThreadLocalActiveSpanSource;
+import org.junit.Test;
+
+import java.util.List;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+
+import static io.opentracing.examples.TestUtils.assertSameTrace;
+import static io.opentracing.examples.TestUtils.sleep;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNull;
+
+public class TestLateSpanFinish {
+
+    private final MockTracer tracer = new MockTracer(new ThreadLocalActiveSpanSource(),
+            Propagator.TEXT_MAP);
+    private final ExecutorService executor = Executors.newCachedThreadPool();
+
+    @Test
+    public void test() throws Exception {
+    /* Create a Span manually and use it as parent of a pair of subtasks */
+        Span parentSpan = tracer.buildSpan("parent").startManual();
+        submitTasks(parentSpan);
+
+    /* Wait for the threadpool to be done first, instead of polling/waiting */
+        executor.shutdown();
+        executor.awaitTermination(15, TimeUnit.SECONDS);
+
+    /* Late-finish the parent Span now */
+        parentSpan.finish();
+
+        List<MockSpan> spans = tracer.finishedSpans();
+        assertEquals(3, spans.size());
+        assertEquals("task1", spans.get(0).operationName());
+        assertEquals("task2", spans.get(1).operationName());
+        assertEquals("parent", spans.get(2).operationName());
+
+        assertSameTrace(spans);
+
+        assertNull(tracer.activeSpan());
+    }
+
+    /* Fire away a few subtasks, passing a parent Span whose lifetime
+     * is not tied at-all to the children */
+    private void submitTasks(final Span parentSpan) {
+
+        executor.submit(new Runnable() {
+            @Override
+            public void run() {
+                // Alternative to calling makeActive() is to pass it manually to asChildOf() for each created Span.
+                try (ActiveSpan span = tracer.makeActive(parentSpan)) {
+                    try (ActiveSpan childSpan1 = tracer.buildSpan("task1").startActive()) {
+                        sleep(55);
+                    }
+                    span.capture(); // Workaround, prevent parentSpan from being finished here.
```

Comment:
The idea for this test sample was to have a 'master' Span whose lifetime is not tied to any children - hence we couldn't pass a `Continuation`, as creating them would increment the refcount for them.

Then again, another option to prevent this usage would be to simply have them create a `Continuation` and don't close the 'master' `ActiveSpan` (which, I think, you will definitely prefer ;) )

Reply:
These examples need more explanatory comments. Especially for this anti-patterns, explain **why** it is written like it is.